@abyesilyurt Could you help fix the few remaining ones? Thanks!

```
packages/syft/tests/syft/service/sync/sync_resolve_single_test.py:90:16: B011 Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`
packages/syft/tests/syft/stores/kv_document_store_test.py:47:10: B017 `pytest.raises(Exception)` should be considered evil
packages/syft/tests/syft/stores/kv_document_store_test.py:82:10: B017 `pytest.raises(Exception)` should be considered evil
tests/integration/local/twin_api_sync_test.py:191:16: B011 Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -78,6 +78,7 @@ repos:
 
   - repo: https://github.com/nbQA-dev/nbQA
     rev: 1.8.3
+    files: "^notebooks/(api|tutorials|admin)"
     hooks:
       - id: nbqa-isort
       - id: nbqa-black

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -84,7 +84,7 @@ repos:
 
   - repo: https://github.com/astral-sh/ruff-pre-commit
     # Ruff version.
-    rev: "v0.3.0"
+    rev: "v0.4.3"
     hooks:
       - id: ruff
         args: [--fix, --exit-non-zero-on-fix, --show-fixes]

--- a/packages/syft/src/syft/service/api/api_service.py
+++ b/packages/syft/src/syft/service/api/api_service.py
@@ -308,8 +308,8 @@ class APIService(AbstractService):
             context.credentials,
             method,
             path,
-            worker_pool=custom_endpoint.worker_pool,
             *args,
+            worker_pool=custom_endpoint.worker_pool,
             **kwargs,
         )
         if isinstance(result, SyftError):
@@ -335,7 +335,10 @@ class APIService(AbstractService):
             time.sleep(0.1)
             if (time.time() - custom_endpoint.endpoint_timeout) > start:
                 return SyftError(
-                    message=f"Function timed out in {custom_endpoint.endpoint_timeout} seconds. Get the Job with id: {job_id} to check results."
+                    message=(
+                        f"Function timed out in {custom_endpoint.endpoint_timeout} seconds. "
+                        + f"Get the Job with id: {job_id} to check results."
+                    )
                 )
 
         if job.status == JobStatus.COMPLETED:

--- a/packages/syft/tests/syft/notebook_ui_test.py
+++ b/packages/syft/tests/syft/notebook_ui_test.py
@@ -61,8 +61,6 @@ def test_sort_table_rows():
     table_data, _ = prepare_table_data(users)
 
     # No sorting
-    for i, row in enumerate(table_data):
-        print(row)
     table_emails = [row["email"] for row in table_data]
     table_indices = [row[TABLE_INDEX_KEY] for row in table_data]
     assert table_emails == emails

--- a/packages/syft/tests/syft/stores/kv_document_store_test.py
+++ b/packages/syft/tests/syft/stores/kv_document_store_test.py
@@ -45,7 +45,7 @@ def test_kv_store_partition_init_failed(root_verify_key) -> None:
     settings = PartitionSettings(name="test", object_type=MockObjectType)
 
     with pytest.raises(Exception):
-        kv_store_partition = KeyValueStorePartition(
+        KeyValueStorePartition(
             UID(), root_verify_key, settings=settings, store_config=store_config
         )
 
@@ -80,7 +80,7 @@ def test_kv_store_partition_set_backend_fail(root_verify_key) -> None:
     settings = PartitionSettings(name="test", object_type=MockObjectType)
 
     with pytest.raises(Exception):
-        kv_store_partition = KeyValueStorePartition(
+        KeyValueStorePartition(
             UID(), root_verify_key, settings=settings, store_config=store_config
         )
 

